pkg/instance/handler.go

```diff
@@ -167,6 +156,44 @@ func (h Handler) Deploy(c *gin.Context) {
 	c.JSON(http.StatusAccepted, savedInstance)
 }
 
+func (h Handler) consumeParameters(user *models.User, sourceInstanceId uint, instance *model.Instance, preset bool) error {
+	sourceInstance, err := h.instanceService.FindByIdDecrypted(sourceInstanceId)
+	if err != nil {
+		return err
+	}
+
+	if sourceInstance.Preset != preset {
```

Comment:
Only to make it easier to read I would write this as

`if preset && sourceInstance.Preset != false`

the next one

`if preset && sourceInstance.StackName != instance.StackName`

if its only used for validating the preset request (which judging the error message it is).